From 5256a970bfbeaf6034bb04335908f6980c728df2 Mon Sep 17 00:00:00 2001
From: shengfei Xu <xsf@rock-chips.com>
Date: Thu, 24 Aug 2017 10:22:54 +0800
Subject: [PATCH] regulator: rk808: fix the enable_val of rk816

The bitmask value must equal enable_val, then the regulator can
be enabled with the regulator output at the predefined voltage.

Change-Id: Ieadac80c04f3826b364d6fd9fa2e3c956f79b6c4
Signed-off-by: shengfei Xu <xsf@rock-chips.com>
---
 drivers/regulator/rk808-regulator.c | 31 ++++++++++++++++++-----------
 1 file changed, 19 insertions(+), 12 deletions(-)

diff --git a/drivers/regulator/rk808-regulator.c b/drivers/regulator/rk808-regulator.c
index 80ae4c793eec..6e0619903add 100644
--- a/drivers/regulator/rk808-regulator.c
+++ b/drivers/regulator/rk808-regulator.c
@@ -675,7 +675,7 @@ static const struct regulator_desc rk816_reg[] = {
 		.vsel_mask = RK818_BUCK_VSEL_MASK,
 		.enable_reg = RK816_DCDC_EN_REG1,
 		.enable_mask = BIT(4) | BIT(0),
-		.enable_val = BIT(0),
+		.enable_val = BIT(4) | BIT(0),
 		.disable_val = BIT(4),
 		.owner = THIS_MODULE,
 	}, {
@@ -693,7 +693,7 @@ static const struct regulator_desc rk816_reg[] = {
 		.vsel_mask = RK818_BUCK_VSEL_MASK,
 		.enable_reg = RK816_DCDC_EN_REG1,
 		.enable_mask = BIT(5) | BIT(1),
-		.enable_val = BIT(1),
+		.enable_val = BIT(5) | BIT(1),
 		.disable_val = BIT(5),
 		.owner = THIS_MODULE,
 	}, {
@@ -707,7 +707,7 @@ static const struct regulator_desc rk816_reg[] = {
 		.n_voltages = 1,
 		.enable_reg = RK816_DCDC_EN_REG1,
 		.enable_mask = BIT(6) | BIT(2),
-		.enable_val =  BIT(2),
+		.enable_val =  BIT(6) | BIT(2),
 		.disable_val = BIT(6),
 		.owner = THIS_MODULE,
 	}, {
@@ -725,13 +725,14 @@ static const struct regulator_desc rk816_reg[] = {
 		.vsel_mask = RK818_BUCK4_VSEL_MASK,
 		.enable_reg = RK816_DCDC_EN_REG1,
 		.enable_mask = BIT(7) | BIT(3),
-		.enable_val = BIT(3),
+		.enable_val = BIT(7) | BIT(3),
 		.disable_val = BIT(7),
 		.owner = THIS_MODULE,
 	},
 	RK816_DESC(RK816_ID_BOOST, "DCDC_BOOST", "boost", 4700, 5400, 100,
 		   RK816_BOOST_ON_VESL_REG, RK818_BOOST_ON_VSEL_MASK,
-		   RK816_DCDC_EN_REG2, BIT(4) | BIT(0), BIT(0), BIT(4), 0),
+		   RK816_DCDC_EN_REG2, BIT(4) | BIT(0), BIT(4) | BIT(0),
+		   BIT(4), 0),
 	{
 		.name = "OTG_SWITCH",
 		.supply_name = "usb",
@@ -743,28 +744,34 @@ static const struct regulator_desc rk816_reg[] = {
 		.n_voltages = 1,
 		.enable_reg = RK816_DCDC_EN_REG2,
 		.enable_mask = BIT(6) | BIT(2),
-		.enable_val = BIT(2),
+		.enable_val = BIT(6) | BIT(2),
 		.disable_val = BIT(6),
 		.owner = THIS_MODULE,
 	},
 	RK816_DESC(RK816_ID_LDO1, "LDO_REG1", "vcc5", 800, 3400, 100,
 		   RK816_LDO1_ON_VSEL_REG, RK818_LDO_VSEL_MASK,
-		   RK816_LDO_EN_REG1, BIT(4) | BIT(0), BIT(0), BIT(4), 400),
+		   RK816_LDO_EN_REG1, BIT(4) | BIT(0), BIT(4) | BIT(0),
+		   BIT(4), 400),
 	RK816_DESC(RK816_ID_LDO2, "LDO_REG2", "vcc5", 800, 3400, 100,
 		   RK816_LDO2_ON_VSEL_REG, RK818_LDO_VSEL_MASK,
-		   RK816_LDO_EN_REG1, BIT(5) | BIT(1), BIT(1), BIT(5), 400),
+		   RK816_LDO_EN_REG1, BIT(5) | BIT(1), BIT(5) | BIT(1),
+		   BIT(5), 400),
 	RK816_DESC(RK816_ID_LDO3, "LDO_REG3", "vcc5", 800, 3400, 100,
 		   RK816_LDO3_ON_VSEL_REG, RK818_LDO_VSEL_MASK,
-		   RK816_LDO_EN_REG1, BIT(6) | BIT(2), BIT(2), BIT(6), 400),
+		   RK816_LDO_EN_REG1, BIT(6) | BIT(2), BIT(6) | BIT(2),
+		   BIT(6), 400),
 	RK816_DESC(RK816_ID_LDO4, "LDO_REG4", "vcc6", 800, 3400, 100,
 		   RK816_LDO4_ON_VSEL_REG, RK818_LDO_VSEL_MASK,
-		   RK816_LDO_EN_REG1, BIT(7) | BIT(3), BIT(3), BIT(7), 400),
+		   RK816_LDO_EN_REG1, BIT(7) | BIT(3), BIT(7) | BIT(3),
+		   BIT(7), 400),
 	RK816_DESC(RK816_ID_LDO5, "LDO_REG5", "vcc6", 800, 3400, 100,
 		   RK816_LDO5_ON_VSEL_REG, RK818_LDO_VSEL_MASK,
-		   RK816_LDO_EN_REG2, BIT(4) | BIT(0), BIT(0), BIT(4), 400),
+		   RK816_LDO_EN_REG2, BIT(4) | BIT(0), BIT(4) | BIT(0),
+		   BIT(4), 400),
 	RK816_DESC(RK816_ID_LDO6, "LDO_REG6", "vcc6", 800, 3400, 100,
 		   RK816_LDO6_ON_VSEL_REG, RK818_LDO_VSEL_MASK,
-		   RK816_LDO_EN_REG2, BIT(5) | BIT(1), BIT(1), BIT(5), 400),
+		   RK816_LDO_EN_REG2, BIT(5) | BIT(1), BIT(5) | BIT(1),
+		   BIT(5), 400),
 };
 
 static const struct regulator_desc rk818_reg[] = {
-- 
2.35.3

